frontend/src/metabase/admin/settings/app/components/SettingsEditor/tests/embedding/premium.embedding.unit.spec.tsx

```diff
@@ -511,7 +513,193 @@ describe("[EE, with token] embedding settings", () => {
 
     expect(screen.queryByText("Static embedding")).not.toBeInTheDocument();
     expect(
-      screen.queryByText("Embedding SDK for React"),
+      screen.queryByText("Embedded analytics SDK"),
     ).not.toBeInTheDocument();
   });
+
+  describe("self-hosted (pro)", () => {
+    beforeEach(async () => {
+      await setupPremium({
+        isHosted: false,
+        settingValues: { "is-hosted?": false },
+      });
+
+      // Go to embedding SDK settings page
+      await userEvent.click(
+        within(
+          screen.getByRole("article", {
+            name: "Embedded analytics SDK",
+          }),
+        ).getByRole("button", { name: "Configure" }),
+      );
+    });
+
+    describe("Embedding SDK settings page", () => {
+      it("should show API key banner", () => {
+        const apiKeyBanner = screen.getByText(
+          /You can test Embedded analytics SDK/,
+        );
+        expect(apiKeyBanner).toHaveTextContent(
+          "You can test Embedded analytics SDK on localhost quickly by using API keys. To use the SDK on other sites, implement JWT SSO.",
+        );
+
+        const withinApiKeyBanner = within(apiKeyBanner);
+        expect(
+          withinApiKeyBanner.getByRole("link", {
+            name: "implement JWT SSO",
+          }),
+        ).toHaveProperty(
+          "href",
+          "https://www.metabase.com/learn/metabase-basics/embedding/securing-embeds?utm_source=product&utm_medium=docs&utm_campaign=embedding-sdk&utm_content=embedding-sdk-admin&source_plan=pro-self-hosted",
+        );
+      });
+
+      it("should show quick start section", () => {
+        expect(screen.getByText("Get started")).toBeInTheDocument();
+        expect(
+          screen.queryByText("Use the SDK with API keys for development."),
+        ).not.toBeInTheDocument();
+
+        expect(
+          screen.getByRole("link", { name: "Check out the Quick Start" }),
+        ).toHaveProperty(
+          "href",
+          "https://metaba.se/sdk-quick-start?utm_source=product&utm_medium=docs&utm_campaign=embedding-sdk&utm_content=embedding-sdk-admin&source_plan=pro-self-hosted",
+        );
+      });
+
+      it("should show CORS settings", () => {
+        expect(
+          screen.getByLabelText("Cross-Origin Resource Sharing (CORS)"),
+        ).toBeInTheDocument();
+
+        expect(
+          screen.getByText(
+            "Enter the origins for the websites or apps where you want to allow SDK embedding, separated by a space. Localhost is automatically included.",
```

Comment:
I deliberately not including the spec part. There was a thread about not using frame-ancestors spec because they're technically not the same. I couldn't find the thread now.